lib/entice/logic/skills/skill.ex

```diff
@@ -92,8 +96,31 @@ defmodule Entice.Logic.Skill.Behaviour do
   defcallback apply_effect(target_entity_id :: term, caster_entity :: %Entity{}) ::
     :ok |
     {:error, reason :: term}
+
+  @doc "Is called before starting to cast."
+  defcallback prerequisites_fulfilled?(target_entity_id :: term, caster_entity :: %Entity{}) ::
+    :ok |
+    {:error, reason :: term}
 end
 
+defmodule Entice.Logic.Skill.Prerequisite do
+  @moduledoc """
+  Helpers that can be used when implementing skill prerequisite scripts
+  """
+  use Entice.Logic.Attributes
+  alias Entice.Entity
+
+  def target_dead?(target_id) when is_binary(target_id) do
```

Comment:
same as with the naimg of the other function, questionmarks are reserved for functions returning boolean by convention, so i'd suggest to opt for 'require_dead(...)' 'require_alive(...)' etc - what do you think?